### PR TITLE
Remove knife-container to use chef-provisioning-docker and node_js improvements

### PR DESCRIPTION
**Pull request by PierreRambaud**
- Add Chef-provisioning-docker
- Remove Ubuntu 12.04 in kitchen.yml
- Allow usage of ChefDk in Rakefile
- Use binaries Ruby 2.1.0  due to readline errors (CF:Rails cookbook)
- Use Xvfb and Google chrome for Selenium tests (node_js)

### CODE REVIEW

**Comment by Sliim** [comment]
It could be cool to keep the `Vagrantfile`, not necessary with the `builder` recipe but simply to provide an easy way to deploy a vm configured with all requirements. No?

Also, I think we can move some configuration, such as recipes inclusion and attributes settings, from recipes to machine's resources in the `containers/` directory, for example:

Instead of have this in `recipes/rbenv.rb`:
```
node.default['rbenv']['user'] = 'root'
node.default['rbenv']['group'] = 'root'
node.default['rbenv']['user_home'] = '/root/'

include_recipe 'container'
include_recipe 'rbenv'
include_recipe 'rbenv::ruby_build'
```
We could have this in `containers/ruby.rb`:
```
machine :ruby do
  machine_options docker_options: {...}

  recipe 'container'
  recipe 'rbenv'
  recipe 'rbenv::ruby_build'
  recipe 'container::rbenv'

  attributes 'rbenv' => {
                    'user' => 'root',
                    'group' => 'root',
                    'user_home' => '/root/'
                  }
end
```

And same for attributes, I think we shouldn't continue to override attributes in our `recipes/` or `attributes/` directories and set them in machine's resources (or a role for common configuration) which is a better place since we use `chef-provisioning`. What do you think about that?

Thanks :)

File: .chef/knife.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# See http://docs.opscode.com/config_rb_knife.html
+# for more information on knife configuration options
+
+current_dir = File.dirname(__FILE__)
+
+log_level :info
+log_location STDOUT
+node_name 'nodienode'
```
Comment (Sliim) [inline .chef/knife.rb]: Why `nodienode`?

**Comment by PierreRambaud** [comment]
Great idea! I moved all attributes and some `recipes` into `roles` directory. 
`Vagrantfile` is back, using `chef-zero`, `docker` cookbook and  install dependencies via `bundler`.

**Comment by Sliim** [comment]
I'm not sure it was necessary to create a role per machine, the `machine` resource is not sufficient?
Thanks for the `Vagrantfile`!

**Comment by PierreRambaud** [comment]
Not sure, with role it's easier to regroup all features in one container and tests them with kitchen.

**Comment by Sliim** [comment]
hmm ok.. I will do some tests and merge it today if it's ok.

**Comment by Sliim** [comment]
Ok, can you rebase your code? There is some missing comments for added ruby files but I will add them later.

**Comment by Sliim** [comment]
Merged! Thanks :)

**Pull request merged**